### PR TITLE
Added "reset arenaseason" and "reset honorworld" commands

### PR DESCRIPTION
**Pull request by Helias**
Added "reset arenaseason" and "reset honorworld" commands.

I took this commands from here: https://github.com/WarHead/uwom-server/blob/691b62a0054324e1e57405ff827c4dc6393d3bec/src/server/scripts/Commands/cs_reset.cpp
and I modified and translated it.

Now the commands do the following things:

**.reset arenaseason**
reset all the arena team rating and the seasonGames/Wins and weekGames/Wins at the default value
and with the [delete] parameter delete also all the arena teams. This command works only without arena season inProgress (you can modify it by conf)

**.reset honorworld**
reset at the default values the honorpoints and with the parameter [kills] also the kills

They works but:
.reset honorworld works only for non online players
To see the results of both commands the server needs a restart because the commands modify some tables that we can't reload just with the command .reload name_table so we need to restart the server

### CODE REVIEW

**Comment by ShinDarth** [comment]
If there is a way to reload following tables:

```
`arena_team` 
`arena_team_member`
`character_arena_stats`
```

it would be nice to have that reload performed after ```.reset arenaseason```

**Comment by Helias** [comment]
Any news? is there anything I should do in order to get it merged?

**Comment by tkrokli** [comment]
I don't know the actual answer, but guessing from that Aokromes changed the label [Feedback-PatchFix] to [Feedback-FixOutdatedMissingWIP] after ShinDarth's comment about the possibility of reloading those 3 tables after `.reset arenaseason`, you could comment if that has been looked into and if your latest commit update was a response to that comment or not.

File: src/server/scripts/Commands/cs_reset.cpp [hunk 1]
```diff
@@ -54,6 +56,44 @@ class reset_commandscript : public CommandScript
         return commandTable;
     }
 
+    static bool HandleResetArenaSeasonCommand(ChatHandler* /*handler*/, const char* args)
+    {
+        if (!sWorld->getBoolConfig(CONFIG_ARENA_SEASON_IN_PROGRESS))
```
Comment (jackpoz) [inline src/server/scripts/Commands/cs_reset.cpp]: some kind of info should be provided when executing this command and arena season in progress

**Comment by jackpoz** [comment]
@ShinDarth is https://github.com/TrinityCore/TrinityCore/pull/15679#issuecomment-147068332 needed for a particular reason ? Or without reloading those tables the arena status is kinda bugged ?

File: src/server/scripts/Commands/cs_reset.cpp [hunk 2]
```diff
@@ -54,6 +56,44 @@ class reset_commandscript : public CommandScript
...
+        return true;
+    }
+
+    static bool HandleResetHonorWorldCommand(ChatHandler* /*handler*/, const char* args)
+    {
+        SQLTransaction trans = CharacterDatabase.BeginTransaction();
+        trans->PAppend("UPDATE `characters` SET `totalHonorPoints`=%u,`todayHonorPoints`=0,`yesterdayHonorPoints`=0", sWorld->getIntConfig(CONFIG_START_HONOR_POINTS));
+
+        std::string param = args;
+        if (*args && param == "kills")
```
Comment (jackpoz) [inline src/server/scripts/Commands/cs_reset.cpp]: please use http://www.cplusplus.com/reference/cstring/strcmp/ to compare const char* strings

**Comment by ShinDarth** [comment]
I guess those tables should be reloaded, but I didn't test it. Having a reload command would be good in any case

**Comment by Aokromes** [comment]
This branch has conflicts that must be resolved
 Use the command line to resolve conflicts before continuing.

**Comment by tkrokli** [comment]
`Helias:arenaseason_honorworld` is currently 456 commits behind `TrinityCore:3.3.5` - rebase needed.

**Comment by Helias** [comment]
Thx @tkrokli I'll do it, but I can't do it forever.. if trinitycore don't merge this merge-conflict there will be again

**Comment by tkrokli** [comment]
I totally understand that. The conflicts will not be that often, so next time would be several weeks, if not months, from now on. Just see if rebase will be enough this time and then see when you have time to answer the questions from jackpoz at a later time.

**Comment by Nyeriah** [comment]
There are two comments not addressed https://github.com/TrinityCore/TrinityCore/pull/15679#discussion_r48874029 and https://github.com/TrinityCore/TrinityCore/pull/15679#discussion_r48874469

**Comment by Aokromes** [comment]
The changes to commands are really stranges, on this case it was to fix a crash.

**Comment by jackpoz** [comment]
> To see the results of both commands the server needs a restart

this makes the commands very misleading, a user would think they actually reset at runtime what they are supposed to reset.

**Comment by Helias** [comment]
> this makes the commands very misleading, a user would think they actually reset at runtime what they are supposed to reset.

I know, but I don't know how to improve this.

> Also, Shauren made an important note that apparently got lost in a previous PR: any changes to the character tables will be overwritten the next time they're saved if they're online. Currently there's no documentation about it in the command description. Since I don't think that's quite clear for the casual users I think it should be added to the command's help, or something, if you really don't plan on addressing it some other way.

I will add something about when and how to use the command in the help description

**Comment by Aokromes** [comment]
@Helias
This branch has conflicts that must be resolved
Use the command line to resolve conflicts before continuing.

If Helias don't answers, anyone can takeover this work?

**Comment by jackpoz** [comment]
There are still some unresolved comments dating back the 5th of January

**Comment by jackpoz** [comment]
> Is there a plan to implement this part too ?

it seems there's no plan

**Pull request closed without merging**